rllib/algorithms/algorithm_config.py

```diff
@@ -443,6 +443,7 @@ def __init__(self, algo_class=None):
         self.ope_split_batch_by_episode = True
         self.evaluation_num_workers = 0
         self.custom_evaluation_function = None
+        self.custom_async_evaluation_function = None
```

Comment:
Dumb question: Why do we need a new config property? Shouldn't we just re-use the existing one? `custom_evaluation_function`. It's implicit that when providing a `custom_evaluation_function` AND the settings are `enable_async_evaluation=True`.

Reply:
I fully agree. I introduced this one in regard to the validations thzat took place and in regard to signal that this function should be asynchronous - I guess there will be no sychronous evaluation anymore?

Reply:
Yes, let's remove this. I think it's obvious for users that if enable_async_evaluation is True and a custom function is provided, that this function will be called async by RLlib.
I mean it doesn't require the python async decorator or anything. It'll simply be sent to the remote workers via the ActorManagers foreach_actor_async method. There is nothing specific that the function has to do differently from a synchronous custom eval function, correct?

Reply:
Let's remove this property and make sure we are always using the `config.custom_eval_function` one.

No, we won't get rid of synchronous evaluation, but a user can only ever use one type (syc or async) in their setup.

Reply:
Ok, I understand now. Let's keep this naming and new property b/c this should be a completely different type of function (having to handle the asynchronoucy of sending these requests to the eval workers itself).